### PR TITLE
Fix for #3640: Setting Null values in query rather than passing them as query parameters for AbstractSqliteDriver

### PR DESCRIPTION
**Pull request by velval**
This fixes issues with typeorm when using with NativeScript in Android platform. Underlying sqlite driver for Android requires all query parameter not to be null, so when updating an existing column value with null typeorm should set it directly into the query rather than passing null as one of the query parameters.

https://github.com/typeorm/typeorm/issues/3640

### CODE REVIEW

**Comment by webleaf** [comment]
@velval there are [tslint errors](https://travis-ci.org/typeorm/typeorm/jobs/513914057)

**Comment by velval** [comment]
Hey @webleaf , I've fixed those tslint errors and all checks seemed have passed now.

File: src/query-builder/UpdateQueryBuilder.ts [hunk 1]
```diff
@@ -388,6 +388,10 @@ export class UpdateQueryBuilder<Entity> extends QueryBuilder<Entity> implements
                     // todo: duplication zone
                     if (value instanceof Function) { // support for SQL expressions in update query
                         updateColumnAndValues.push(this.escape(column.databaseName) + " = " + value());
+                    } else if ( (value === undefined || value === null) && this.connection.driver instanceof AbstractSqliteDriver ) {
```
Comment (pleerock) [inline src/query-builder/UpdateQueryBuilder.ts]: `undefined !== null`. Why do you check for undefined value here?

File: src/query-builder/UpdateQueryBuilder.ts [hunk 2]
```diff
@@ -434,6 +438,10 @@ export class UpdateQueryBuilder<Entity> extends QueryBuilder<Entity> implements
                 // todo: duplication zone
                 if (value instanceof Function) { // support for SQL expressions in update query
                     updateColumnAndValues.push(this.escape(key) + " = " + value());
+                } else if ( (value === undefined || value === null) && this.connection.driver instanceof AbstractSqliteDriver ) {
```
Comment (pleerock) [inline src/query-builder/UpdateQueryBuilder.ts]: same here

**Comment by webleaf** [comment]
I think we could do `value == null`. It checks for `undefined` too.
But (even in this case) set entity field to `undefined` does not produce `UPDATE` query either (looks like it by design in typeorm)
About the fix as a whole, it seems to work for me.

**Comment by pleerock** [comment]
Im not asking to do `value == null`, Im asking why we need to do more than `value === null`.

> But (even in this case) set entity field to undefined does not produce UPDATE query either (looks like it by design in typeorm)

right, this is why Im asking why this undefined check is needed here

**Comment by webleaf** [comment]
@pleerock please, make as it would be better

**Comment by jscti** [comment]
Can this be merged ?

**Review by imnotjames — CHANGES_REQUESTED** [review]
If this is only affecting native script, why update all SQLite drivers?

**Comment by imnotjames** [comment]
Given there were unanswered questions, last activity over a year ago & this would negatively impact other drivers I'll be closing it.

I'll be closing this PR but if you ever get time to come back to it please feel welcome to open a new PR addressing the concerns.

**Pull request closed without merging**